operator/pkg/resources/statefulset_update.go

```diff
@@ -181,6 +221,14 @@ func (r *StatefulSetResource) getPodList(ctx context.Context) (*corev1.PodList,
 		return nil, fmt.Errorf("unable to list pods: %w", err)
 	}
 
+	podList.Items = slices.DeleteFunc(podList.Items, func(pod corev1.Pod) bool {
```

Comment:
nit: You could just update the LabelSelector to exclude the `NodePoolKey` label instead of performing client side filtering.

Reply:
i would prefer this as well, and attempted to do this initially. 

however, because of how the code in pkg/labels is built, it requires quite some plumbing to have a different selector. a change is required in multiple places; i would prefer to not do a larger refactoring here.

since this is flagged as a nit, i suggest to keep this as-is.